regulations/tasks.py

```diff
@@ -38,6 +47,13 @@ def build_comment(body):
 
 
 def extract_files(body):
+    '''
+    Extracts the files that are to be attached to the comment.
+    Returns a collection of dicts where for each dict:
+        - dict['key'] specifies the file to be attached from S3
+        - dict['name'] specifies the name under which the file is to be
+          attached.
+    '''
```

Comment:
A namedtuple gets you closer to type safety, but I see your point. The extreme comes elsewhere in  -site, where a dictionary is the _primary_ data type (representing tree nodes).